### DISCUSSION TITLE
Casting MultiDiGraph to DiGraph

### DISCUSSION THREAD

**Discussion by rcyost** [networkx/networkx#4731] [Q&A] [answered]
How does this treat weights and attributes?

I think it just takes the values from the last edge in the MDG, but would like to confirm and see if there are ways to aggregates weights or attributes . 

Thanks !

**Comment by dschult** [comment, marked as answer]
It's best to aggregate after casting. Perhaps something like this:
```python
DG = nx.DiGraph(MDG)
for edge in DG.edges:
    DG.edges[edge]["weight"] = sum(wt for u, v, wt in MDG.edges(edge, data="weight"))
```
If you have many attributes to process, you probably want to get the edge keydict and accumulate over that.
```python
DG = nx.DiGraph(MDG)
for u, v in DG.edges:
    keydict = MDG[u][v]
    DG.edges[u, v]["weight"] = sum(dd['weight'] for dd in keydict.values())
    DG.edges[u, v]["capacity"] = max(dd['capacity'] for dd in keydict.values())
```

**Reply by rcyost** [reply]
Hello, thanks for your help. What should edge be defined with in row 4,5 of the second example?

**Reply by dschult** [reply]
It was a typo.  Fixed now.  I copy/pasted from previous example where we didn't need to split the edge tuple into nodes.

**Reply by rcyost** [reply]
Thank you so much, this is very helpful!
